### PR TITLE
refactor!: remove supportsInterface

### PR DESCRIPTION
**Pull request by fhildeb**
### What kind of change does this PR introduce?

**Braking Feature Change:** This PR removes the `this.supportsInterface(...)` and `ERC725.supportsInterface(...)` functions, as well as their tests and documentation.

### What is the current behavior?

Currently, erc725.js can be used to resolve and detect whitelisted [ERC165-based](https://eips.ethereum.org/EIPS/eip-165) interface IDs of a given contract based on their standardized string inputs:

```js
// Object
myErc725.supportsInterface("ERC725X")
myErc725.supportsInterface("LSP8IdentifiableDigitalAsset")

// Class
ERC725.supportsInterface("ERC725X", "0x...")
ERC725.supportsInterface("LSP8IdentifiableDigitalAsset", "0x...")
``` 

While this is a convenient way to check interfaces without further redo, resolving and manually updating IDs can be cumbersome. 

### What is the new behavior?

The decision was made to completely remove this functionality and keep it completely separated from contract libraries (like [lsp-smart-contracts](https://docs.lukso.tech/tools/lsp-smart-contracts/getting-started)), that feature additional information like ABIs and schemas on top of being up-to-date with the latest standardization changes.

If you are affected by this **breaking feature change**, please use the following method to read ERC165-based interface IDs:

```js
// Example using IDs from the @lukso/lsp-smart-contracts library
import { INTERFACE_IDS } from "@lukso/lsp-smart-contracts";

// Define a global EIP165 ABI 
const eip165ABI = [{
    inputs: [{
        internalType: 'bytes4',
        name: 'interfaceId',
        type: 'bytes4',
      }],
    name: 'supportsInterface',
    outputs: [{
        internalType: 'bool',
        name: '',
        type: 'bool',
      }],
    stateMutability: 'view',
    type: 'function',
  }];
  
  try {
    const myEip165Contract = contract(eip165ABI as any, address)
    return await eip165Contract.methods.supportsInterface(INTERFACE_IDS.ERC725X).call()
    return await eip165Contract.methods.supportsInterface(INTERFACE_IDS.LSP8IdentifiableDigitalAsset).call()
  } catch (error: unknown) {
    console.error(error)
    return false
  }
```

### Other information:

The following repositories have been checked:
✅ [erc725-inspect](https://erc725-inspect.lukso.tech/)
✅ [my.universalprofile.cloud](https://my.universalprofile.cloud/)
✅ [wallet.universalprofile.cloud](https://wallet.universalprofile.cloud/)

None of them is using ERC725's `supportsInterface` function anymore.

### CODE REVIEW

**Comment by CJ42** [comment]
This could be duplicate with #317
I had opened the PR already, but we could close it and keep this one (or the other way around)

**Comment by fhildeb** [comment]
Oh you are right. Did not check if there already was a PR open for that. 😅 Has there been an outcome yet? @frozeman

**Comment by CJ42** [comment]
From my side, I would go ahead with this PR so that we focus this library purely on the encoding / decoding side.
Which is what this library is aimed for.

**Comment by Hugoo** [comment]
Nice that the other repos were updated.

If this is a breaking change, don't forget to add `!` in the commit name.

```
refactor: remove supportsInterface
```

https://www.conventionalcommits.org/en/v1.0.0/

File: src/lib/detector.ts [hunk 1]
```diff
@@ -16,47 +16,11 @@
 
 /**
  * @file detector.ts
- * @author Hugo Masclet <@Hugoo>
- * @author Felix Hildebrandt <@fhildeb>
```
Comment (Hugoo) [inline src/lib/detector.ts:20]: ??

File: src/lib/detector.test.ts [hunk 2]
```diff
@@ -13,56 +13,14 @@
 */
 /**
  * @file lib/detector.test.ts
- * @author Hugo Masclet <@Hugoo>
```
Comment (Hugoo) [inline src/lib/detector.test.ts:16]: Please keep

File: src/lib/detector.ts (hunk 1 above)
Comment (fhildeb) [reply src/lib/detector.ts:20]: The file was created specifically for the supportsInterface. All other code comes from Yamen (permissions)

File: src/lib/detector.test.ts (hunk 2 above)
Comment (fhildeb) [reply src/lib/detector.test.ts:16]: as above, non of your/our your code is left in this file after removing this function.

**Review by Hugoo — APPROVED** [review]

**Pull request merged**